### PR TITLE
added assertion_utils

### PR DESCRIPTION
**Pull request by dwl285**
keyUniqueness if the only function for now!

### CODE REVIEW

File: assertion_utils/BUILD [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package(default_visibility = ["//visibility:public"])
+
+load("@npm_bazel_typescript//:index.bzl", "ts_library")
+
+ts_library(
+    name = "assertion_utils",
+    srcs = glob(["**/*.ts"]),
+    module_name = "@dataform/assertion_utils",
+    deps = [
+        "//core",
```
Comment (BenBirt) [inline assertion_utils/BUILD]: nit: rm this, you don't actually have this dep

File: assertion_utils/assertion_utils.package.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@dataform/assertion_utils",
+  "description": "Assertion utilities for dataform.",
```
Comment (BenBirt) [inline assertion_utils/assertion_utils.package.json]: nit: capitalize Dataform?

File: assertion_utils/index.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+
```
Comment (BenBirt) [inline assertion_utils/index.ts]: nit: rm empty line

File: assertion_utils/index.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+export function forDataset(foo: string) {
```
Comment (BenBirt) [inline assertion_utils/index.ts]: foo => dataset

File: assertion_utils/index.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+export class DatasetAssertion {
+  private readonly dataset: string;
+  private groupCols: string[] = [];
+  constructor(dataset: string) {
+    this.dataset = dataset;
+  }
+
+  public groupedBy(cols: string[]) {
```
Comment (BenBirt) [inline assertion_utils/index.ts]: many of our APIs allow you to pass either a single string or a string array, i would suggest that we do the same here.

```
public groupedBy(cols: string | string[]) {
  this.groupCols = typeof cols === "string" ? [cols] : cols;
  return this;
}
```

File: assertion_utils/index.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    this.groupCols = cols;
+    return this;
+  }
+
+  public getUniqueRowQuery(): string {
+    return `
+    WITH base AS (
```
Comment (BenBirt) [inline assertion_utils/index.ts]: is this bigquery-formatted? might be nice to follow that.

Comment (dwl285) [reply assertion_utils/index.ts]: Opting to ignore this one :)

**Comment by dwl285** [comment]
ready for another review now @BenBirt

**Comment by dwl285** [comment]
Changes made (except bigquery formatting) @BenBirt

**Pull request merged**